CHANGELOG.md

```diff
@@ -1,5 +1,13 @@
 # CHANGELOG
 
+## 0.6.0
+> Published 18 November 2024
+
+### Features
+- On JVM, libraries are now include Java Platform Module System (JPMS) modules [#406](https://github.com/Kotlin/kotlinx-io/pull/406)
+- Made Segment's size and capacity public [#409](https://github.com/Kotlin/kotlinx-io/pull/409)
```

Comment:
I think we should also mention changes from https://github.com/Kotlin/kotlinx-io/pull/414